zen-shell/shell.c

```diff
@@ -127,6 +147,19 @@ zen_shell_bind(
   wl_resource_set_implementation(resource, &shell_interface, shell, NULL);
 }
 
+WL_EXPORT void
+zen_shell_unset_background(struct zen_shell* shell)
+{
+  struct zen_background *background, *tmp;
+
+  wl_list_for_each_safe(background, tmp, &shell->background_list, link)
+  {
+    free(background->virtual_object->role);
+    background->virtual_object->role = strdup("");
+    zen_background_destroy(background);
+  }
```

Comment:
```suggestion
  wl_list_for_each_safe(background, tmp, &shell->background_list, link)
  {
    zen_background_destroy(background);
  }
```

一応僕のsuggestionも入れておきます